src/ng-currency.directive.js

```diff
@@ -119,6 +127,10 @@ export default function ngCurrency($filter, $locale, $timeout, ngCurrencySetting
       function revalidate() {
         controller.$validate();
         if (active) {
+          if (autoFill && [undefined, null, ''].indexOf(controller.$$rawModelValue) !== -1) {
+            controller.$$rawModelValue = '0';
```

Comment:
One thing I noticed with this solution is that the value never gets commited to the `modelValue`.

Instead can you move this logic to the top of the `keepInRange` function. That way the if statement below gets executed.

```js
function keepInRange(value) {
  if (autoFill && [undefined, null, ''].indexOf(value) !== -1) {
    value = 0;
  }
  // ...
}
```